### DISCUSSION TITLE
关于 tool-view 的一点改进想法

### DISCUSSION THREAD

**Discussion by mark9804** [Arondight/Adachi-BOT#570] [Ideas]
现在的 `npm run tool-view` 默认是调用 puppeteer 打开调试页面的，但其实打开的这个浏览器进程[没有加载插件](https://github.com/puppeteer/puppeteer/blob/main/docs/api.md#working-with-chrome-extensions)，所以我还是习惯把 chromium 的链接复制出来粘贴到 chrome 里面打开，实际上一次都没在 chromium 里面调试过

我觉得如果 tool-view 可以给一个“在系统默认浏览器中打开”的参数会方便一些，在这个参数下，通过 terminal 来启动系统默认的浏览器：
```js
const url =`http://localhost:9934/src/views/${name}.html?${new URLSearchParams(param)}`;
const start = (process.platform == 'darwin'? 'open': process.platform == 'win32'? 'start': 'xdg-open');
require('child_process').exec(start + ' ' + url);
```

**Comment by Arondight** [comment]
嗯，怎么方便怎么来，浏览器的可执行文件可以通过 `puppeteer.executablePath()` 拿，不需要 `render.js` 了

**Reply by Arondight** [reply]
另外最好用同步，也可以把那个啥事不做的 promise 删了，你要是不急的话本地先改着，我周末摸了

**Reply by mark9804** [reply]
我不急，我一般都是 chrome 地址栏里面搜历史记录，只有生成新数据的时候才用 view
